resources/js/interfaces/realtime-room-event-json.ts

```diff
@@ -0,0 +1,36 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the GNU Affero General Public License v3.0.
+// See the LICENCE file in the repository root for full licence text.
+
+export default interface RealtimeRoomEventJson {
+  event_type: RealtimeRoomEventType;
+  id: number;
+  playlist_item_id?: number;
+  timestamp: string;
+  user_id?: number;
+}
+
+export type GameStartedEventJson = RealtimeRoomEventJson & {
+  event_detail: GameStartedEventDetail;
+  event_type: 'game_started';
+};
+
+export type RealtimeRoomEventType =
+  | 'game_started'
+  | 'game_aborted'
+  | 'game_completed'
+  | 'host_changed'
+  | 'player_joined'
+  | 'player_kicked'
+  | 'player_left'
+  | 'room_created'
+  | 'room_disbanded'
+  | 'unknown';
+
+export interface GameStartedEventDetail {
+  room_type: RealtimeRoomMatchType;
```

Comment:
isn't this redundant with the one in `room` and will be the same for all events in that room? Or can it be changed midway? In which case it's more like game/playlist item type?

Reply:
yeah, it's kind of a weird one... if anything the one in room seems more like "current game/playlist type" (when in realtime "type") in this case.

eh 🤷‍♀️